3.0/m_conn_require.cpp

```diff
@@ -429,7 +429,7 @@ class ModuleConnRequire : public Module
 	// This matches with 2.0's placement of OnUserInit.
 	void OnSetUserIP(LocalUser* user) CXX11_OVERRIDE
 	{
-		if (user->registered != REG_NONE)
+		if (userdata.get(user))
```

Comment:
You unset this on `OnUserConnect` but there is no guarantee that a user's IP will not change after registration (e.g. via the IRCv3 resume spec)